src/Engines/BaseEngine.cs

```diff
@@ -590,6 +598,12 @@ public virtual bool IsHelp(string input, out ISymbol symbol, out string commandI
 
         public virtual async Task<IEnumerable<Completion>> CompleteMagic(string code, int cursorPos)
         {
+            if (MagicResolver == null)
+            {
+                Logger.LogDebug("Cannot provide completions, as magic resolver was null.");
```

Comment:
Ideally, this should only be called when there's an active completion request (e.g.: the user pressing "Tab" in a notebook), rather than passively as the user types partial inputs. In any case, the magic resolver should be non-null once it loads properly, and the debug log should not be user visible unless opted-in with `IQSHARP_LOG_LEVEL="Debug"`.